packages/calypso-build/webpack/hot-module-replacement.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * External dependencies
+ */
+const webpack = require( 'webpack' );
+
+/**
+ * Return an array of hot module reloading relevant webpack plugin object
+ *
+ * @param  {Boolean}   isDeveloment  whether the code is bundled is in development mode
+ *
+ * @return {Object[]}  hot module reloading webpack plugin object
+ */
+module.exports.plugins = isDeveloment => {
+	return isDeveloment ? new webpack.HotModuleReplacementPlugin() : null;
```

Comment:
This works OK because we're passing `plugins` through `_.compact`. I don't think we should expect that.

We can pass a noop plugin to make this safer 🙂 

```suggestion
	return isDeveloment ? new webpack.HotModuleReplacementPlugin() : ()=>{};
```

Reply:
Sorry, to clarify. Webpack expects plugins to be an array of functions or objects with an `apply` method.

In the Calypso config, they're passed through `_.compact` to drop falsey values. That allows us to do things like `[ !! predicate && new WebpackPlugin(), ]`. Webpack does not like this and we shouldn't expect all consumers to use `_.compact` on their plugins config.